src/main/java/com/joss/conductor/mobile/annotation/PlatformFindByHelper.java

```diff
@@ -0,0 +1,51 @@
+package com.joss.conductor.mobile.annotation;
+
+import com.joss.conductor.mobile.Platform;
+import org.openqa.selenium.By;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+/**
+ * Created on 3/14/18.
+ */
+public final class PlatformFindByHelper {
+
+    //TODO decide whether or not to check for initial current = null
+
+    public static void initIds(Object page, Platform platform) throws Exception {
+        Class current = page.getClass();
+        do {
+            for (Field field : current.getDeclaredFields()) {
+                for (Annotation annotation : field.getDeclaredAnnotations()) {
+
+                    if (annotation instanceof AndroidFindBy || annotation instanceof IOSFindBy) {
+                        if (!field.getType().equals(By.class)) {
+                            throw new ClassCastException(field.getName() + " must be of type By!");
```

Comment:
I wouldn't use ClassCastException here as it's not a checked exception (hence the throws Exception at the top). I would either create a custom checked Exception and have this method use that or just allow this method to go unchecked (which would be my preference in this case... there's very little you can do to recover from this error).

Reply:
In that case lets just create a  `RuntimeException` to provide a helpful message.